mobile/library/cc/engine_builder.h

```diff
@@ -41,8 +41,15 @@ struct NodeLocality {
 class EngineBuilder {
 public:
   EngineBuilder();
-  // This constructor is not compatible with bootstrap mode.
-  EngineBuilder(std::string config_template);
+  // THIS CONSTRUCTOR SHOULD ONLY BE CALLED BY TESTS. Further uses of this constructor are highly
+  // discouraged as this API is likely to go away soon.
+  //
+  // Creates an EngineBuilder from the provided YAML config string. If this constructor is invoked,
+  // then only the config provided in the string will be used to generate the Envoy engine; none of
+  // the builder function invocations will take effect.
+  //
+  // TODO(abeyad): remove the need for this constructor.
+  explicit EngineBuilder(std::string config);
```

Comment:
Should we remove this completely in favor of `setOverrideConfigForTests` to make it more explicit?